utils/data.cpp

```diff
@@ -0,0 +1,250 @@
+#include "data.hpp"
+
+namespace jbrush {
+
+Data::Data(Canvas* theCanvas)
+{
+    mainCanvas = theCanvas;
+    currentDirectory = STARTING_DIRECTORY;
+}
+
+void Data::Load()
+{
+    try {
+        // Variables used to custom-craft descriptive error messages in exception handling
+        QString errorMessage;
+        QTextStream errorStream(&errorMessage);
+
+        // Bring up a panel for the user to select a directory
+        SelectDirectory();
+
+        // If directory is not valid, throw an exception
+        if(currentDirectory == "")
+        {
+            errorStream << "Directory" << currentDirectory << "is not valid";
+            GeneralException(errorMessage).raise();
+        }
+
+        // Get the file at the current directory
+        // and make a text stream to operate on it
+        QFile file(currentDirectory);
+        QTextStream qin(&file);
+
+        if(!file.open(QIODevice::ReadOnly))
+        {
+           errorStream << "File at" << currentDirectory << "was not opened properly";
+           GeneralException(errorMessage).raise();
+        }
+
+        QString type;   // Name identifying the shape read in
+        QList<qint32> dimensions;   // This list is used to load in the dimensions of the shape
+        Shape* toAdd;   // The shape to be constructed and added to the canvas
+
+        qint32 tempInt; // Temporary storage for an integer
+        QChar tempChar; // Temporary storage for a character
+
+        // Set integer base. Speeds things up a little for the text stream
+        qin.setIntegerBase(10);
+
+        // Loop and extract data until we reach the end of the file stream
+        while(!qin.atEnd())
+        {
+            GetTo(qin, TITLE_INTERRUPTOR);
+            qin >> type;
+
+            // Get to next whitespace after leading title
+            GetTo(qin, TITLE_INTERRUPTOR);
+
+            // Continue reading in integers until one is found that is not followed by the designated dimension interruptor
+            do
+            {
+                qin >> tempInt;
+                dimensions.append(tempInt);
```

Comment:
I believe this is a problem. Not the fact that you are appending each dimension into a ``QList`` (which is kind of heresy since we have our Vector 👎 ), but the fact that you aren't reinitializing your ``QList``. The dimensions of the previous shape is still inside the ``QList``.

This also explains the glitch that was happening when trying to load the Polyline after loading Line. 

What I suggest is have all your variables above this ``while()`` loop be stored inside the ``while()`` loop. This way each iteration has a fresh variable.